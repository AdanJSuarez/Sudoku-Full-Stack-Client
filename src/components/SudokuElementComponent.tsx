/** 
 * Adan J. Suarez
 * adanjsuarez@gmail.com
 * Full Stack Home-Project 
 */

import * as React from 'react';

export interface SudokuElementProps {
    row: number;
    column: number
    dataElement: number;
    toggleNumber: any;
}
export interface SudokuElementState {
    currentRow: number;
    currentColumn: number;
    currentDataElement: number;
    toggleOn: boolean;
}
export default class SudokuElementComponent extends React.Component<SudokuElementProps, SudokuElementState> {
    constructor(props: SudokuElementProps) {
        super(props);
        this.state = {
            currentRow: props.row,
            currentColumn: props.column,
            currentDataElement: props.dataElement,
            toggleOn: false
        };
        this.hundleClick = this.hundleClick.bind(this);
    }
    public render() {
        return (
            <button className={'Element'} onClick={this.hundleClick}> { this.state.currentDataElement } </button>
        );
    }
    /**
     * Handle click bubbling up Json object selected with click
     * @private
     * @param {React.MouseEvent<HTMLButtonElement>} e
     * @memberof SudokuElementComponent
     */
    private hundleClick(e: React.MouseEvent<HTMLButtonElement>){
        let emptyElement = {};
        let elementBubbleUp: any;
        let element = {
            number: this.state.currentDataElement,
            row: this.state.currentRow,
            column: this.state.currentColumn
        }
        this.state.toggleOn? this.setState({ toggleOn: false }) : this.setState({ toggleOn: true })
        this.state.toggleOn? elementBubbleUp = emptyElement:elementBubbleUp = element;
        this.props.toggleNumber(elementBubbleUp);
    }
}
