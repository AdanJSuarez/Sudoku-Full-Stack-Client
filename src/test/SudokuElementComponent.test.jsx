import React from 'react';
import Enzyme, {shallow, render, mount} from 'enzyme';
// import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import {createSerializer} from 'enzyme-to-json';
//import { expect } from 'chai';
import SudokuElementComponent from '../components/SudokuElement';

//Set the default serializer for jest to be from enzyme-to-json
//this produce a more easy to read serialized format
expect.addSnapshotSerializer(createSerializer({mode: "deep"}));
Enzyme.configure({adapter: new Adapter()});

describe("Test suit for SudokuElementComponent", ()=>{
    
    it('render correctly shallow', ()=>{
        const wrapper = shallow(
            <SudokuElementComponent row={1} row={1} column={1} dataElement={2} toggledNumber={{number:2, row:1, column:1}} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render correctly render', ()=>{
        const wrapper = render(
            <SudokuElementComponent row={1} row={1} column={1} dataElement={2} toggledNumber={{number:2, row:1, column:1}} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render correctly mount', () => {
        const wrapper = mount(<SudokuElementComponent row={1} row={1} column={1} dataElement={2} toggledNumber={{number:2, row:1, column:1}}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    //TODO: It needs many more test!!
})