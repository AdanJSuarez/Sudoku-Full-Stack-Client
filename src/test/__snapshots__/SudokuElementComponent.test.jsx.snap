// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit for SudokuElementComponent render correctly render 1`] = `
<button
  class="Element"
>
   2 
</button>
`;

exports[`Test suit for SudokuElementComponent render correctly shallow 1`] = `
<button
  className="Element"
  onClick={[Function]}
>
   
  2
   
</button>
`;
